api/v2/Routes/students/masterymapping/index.js

```diff
@@ -0,0 +1,75 @@
+import { Router } from 'express';
+import {
+    getMaxScores,
+    getStudentScores,
+} from '../../../../lib/redisHelper.mjs';
+import ProgressReportData from '../../../../assets/progressReport/CS10.json' assert { type: 'json' };
+
+const router = Router({ mergeParams: true });
+
+function getTopicsFromUser(gradeData) {
+    const topicsTable = {};
+    Object.entries(gradeData).forEach(([assignment, topics]) => {
+        Object.entries(topics).forEach(([topic, score]) => {
+            if (topic in topicsTable) {
+                topicsTable[topic] += +(score ?? 0);
+            } else {
+                topicsTable[topic] = +(score ?? 0);
+            }
+        });
+    });
+    return topicsTable;
+}
+
+async function getMasteryMapping(userTopicPoints, maxTopicPoints) {
+    const numMasteryLevels = ProgressReportData['student levels'].length - 2;
+    Object.entries(userTopicPoints).forEach(([topic, userPoints]) => {
+        const maxAchievablePoints = maxTopicPoints[topic];
+        if (userPoints === 0) {
+            return;
+        }
+        if (userPoints >= maxAchievablePoints) {
+            userTopicPoints[topic] = numMasteryLevels + 1;
+            return;
+        }
+        const unBoundedMasteryLevel =
+            (userPoints / maxAchievablePoints) * numMasteryLevels;
+        if (unBoundedMasteryLevel === numMasteryLevels) {
+            userTopicPoints[topic] = numMasteryLevels;
+        } else if (unBoundedMasteryLevel % 1 === 0) {
+            // Push them over to the next category if they are exactly on the edge.
+            userTopicPoints[topic] = unBoundedMasteryLevel + 1;
+        } else {
+            userTopicPoints[topic] = Math.ceil(unBoundedMasteryLevel);
+        }
+    });
+    const masteryMapping = {};
+    Object.entries(userTopicPoints).forEach(([topic, userPoints]) => {
+        masteryMapping[topic] = {"student_mastery": userPoints, "class_mastery": 0};
```

Comment:
Where does the class_mastery get updated?

Reply:
Our CM does not use the `class_mastery` feature, so I am setting it to 0 as by default. `class_mastery` can be used to change the color of edges in CM based on class proficiency in a concept.

Reply:
Don't we want to use the class mastery feature though?  I thought that this was supposed to be baked into the timeline?